From 90bf0391e0639309afdd3d48057ed701c961ae4e Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 2 Oct 2023 09:42:50 +0200
Subject: [PATCH 351/954] feat(stm32mp2): disable unsupported features

SPE and SVE for non-secure world are not supported on Arm v8.0.
Disable the corresponding flags. This also saves a bit of memory.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git d2fc2bb80cb7f464eee411cd8435da247bc70e8c feat(stm32mp2): disable unsupported features]
Change-Id: I323fb7410393ea9711759be4c47848316fb68860
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/platform.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 185e26dd5a..a5afc3addf 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -14,6 +14,10 @@ CRASH_REPORTING			:=	1
 ENABLE_PIE			:=	1
 PROGRAMMABLE_RESET_ADDRESS	:=	1
 
+# Disable features unsupported in ARMv8.0
+ENABLE_SPE_FOR_NS		:= 0
+ENABLE_SVE_FOR_NS		:= 0
+
 # Enable PSCI v1.0 extended state ID format
 PSCI_EXTENDED_STATE_ID		:= 1
 PSCI_OS_INIT_MODE		:= 1
-- 
2.53.0

